internal/containers/containers.go

```diff
@@ -88,12 +89,27 @@ func (c *Container) GetPort() (string, bool) {
 	return "", false
 }
 
-func (c *Container) getTargetHostname(hostname string) string {
+func (c *Container) getTargetHostname(ingressNetwork, hostname string) string {
 	// return container IP address if defined
 	if len(c.Info.NetworkSettings.IPAddress) > 0 {
 		return c.Info.NetworkSettings.IPAddress
 	}
 
+	if ingressNetwork != "" && c.Info.NetworkSettings.Networks[ingressNetwork] != nil {
+		dns := c.Info.NetworkSettings.Networks[ingressNetwork].DNSNames[0]
+
+		// Check if the service name is available
+		_, err := net.LookupIP(dns)
+
+		if err != nil {
+			// TODO: Improve error handling
```

Comment:
The error handling can certainly be improved.

Maybe `getTargetHostname()` should return `(string, error)` instead?